Add rst documentation for deprecated enums

andreas-bok-sociomantic opened the pull request:
- Add missing rst documentation for drizzle_socket_option_t
- Replace drizzle_socket_owner with drizzle_socket_owner_t

andreas-bok-sociomantic committed:
Add rst documentation for deprecated enums

Add rst documentation for drizzle_socket_option_t
and drizzle_socket_owner_t and the deprecated
drizzle_socket_option and drizzle_socket_owner

ben-palmer-sociomantic commented on docs/api/constants.rst:
```diff
@@ -741,14 +743,42 @@ Connection
+
+   typedef of :c:type:`drizzle_socket_owner_t`
+
+.. c:type:: drizzle_socket_option_t
```
Should this documentation also be required in `5.x.x`?

andreas-bok-sociomantic replied:
Yeah, just checked. It is in this commit 94cff771a4524c3fc4438487202c3a613eecf355 on v5.x.x

ben-palmer-sociomantic commented on docs/api/constants.rst:
```diff
@@ -716,11 +716,12 @@ Connection
       Verify SSL cert
 
-   .. py:data:: DRIZZLE_CAPABILITIES_REMEBER_OPTIONS
+   .. py:data:: DRIZZLE_CAPABILITIES_REMEMBER_OPTIONS
```
I would separate the fixes and the additions in to two commits.

ben-palmer-sociomantic reviewed (commented):
Couple of minor comments.

andreas-bok-sociomantic commented:
Updated based on comments from @ben-palmer-sociomantic 
Also rebased on v5.x.x since the deprecations were added on this branch and not merged into v6.x.x yet

andreas-bok-sociomantic commented:
Moved the commit into changes for v5.x.x to a new PR (#208)

ben-palmer-sociomantic commented:
LGTM
